lib/pbench/test/unit/server/test_file_tree.py

```diff
@@ -181,6 +182,48 @@ def test_duplicate(self, selinux_disabled, server_config, make_logger, tarball):
             tree.create("ABC", source_tarball)
         assert exc.value.tarball == Dataset.stem(source_tarball)
 
+    def test_unpack(
+        self, monkeypatch, selinux_enabled, server_config, make_logger, tarball
+    ):
+        """Test to check the unpacking functionality of the FileTree"""
+        source_tarball, source_md5, md5 = tarball
+
+        def mock_run(args, cwd, stderr):
+            return subprocess.CompletedProcess(
+                args, returncode=0, stdout=b"Some error running tar", stderr=None
+            )
+
+        monkeypatch.setattr("subprocess.run", mock_run)
+        tree = FileTree(server_config, make_logger)
+        tree.create("ABC", source_tarball)
+
+        tarball = tree.find_dataset(md5)
+        tree.unpack(md5)
+
+    def test_bad_unpack(
```

Comment:
How about a third case where `subprocess.run` raises an exception?

Reply:
It is raising an error in this case and also returning a returncode as 1.. Maybe what I can do is.. separate these two cases.

Reply:
You've got `subprocess.run` returning a `CompletedProcess` with a non-0 return code, which causes `unpack` to raise an exception. I'm talking about the additional case where `subprocess.run` raises an exception causing `unpack` to raise an exception. We want to be sure that both cases work as we expect.